lymph/midline.py

```diff
@@ -758,17 +502,70 @@ def likelihood(
         except ValueError:
             return -np.inf if log else 0.
 
-        if len(self.patient_data.loc[(self.patient_data['info', 'tumor', 'midline_extension']==False) | (self.patient_data['info', 'tumor', 'midline_extension']==True)])>0:
+        llh = 0. if log else 1.
+
+        stored_t_stages = set(self.ext.ipsi.diagnose_matrices.keys())
+        provided_t_stages = set(self.ext.ipsi.diag_time_dists.keys())
+        t_stages = list(stored_t_stages.intersection(provided_t_stages))
+        
+        max_t = self.diag_time_dists.max_t
+        llh = 0. if log else 1.
+
+        state_probs_ipsi_nox = self.noext.ipsi._evolve_onestep()
+        state_probs_contra_nox = self.noext.contra._evolve_onestep()
```

Comment:
If this is one step, how come you only need to compute it once?

Reply:
The name of the function is confusing. It calculates every time step at once but does it step by step and not just jumps to the last time step. I have to think of a more suitable name.

Reply:
I see 👍🏻... How about `_evolve_stepwise()`?